source/opt/ir_context.cpp

```diff
@@ -125,12 +142,44 @@ void spvtools::ir::IRContext::ForgetUses(Instruction* inst) {
   if (AreAnalysesValid(kAnalysisDefUse)) {
     get_def_use_mgr()->EraseUseRecordsOfOperandIds(inst);
   }
+  if (AreAnalysesValid(kAnalysisDecorations)) {
+    if (inst->IsDecoration()) {
+      get_decoration_mgr()->RemoveDecoration(inst);
+    }
+  }
 }
 
 void IRContext::AnalyzeUses(Instruction* inst) {
   if (AreAnalysesValid(kAnalysisDefUse)) {
     get_def_use_mgr()->AnalyzeInstUse(inst);
   }
+  if (AreAnalysesValid(kAnalysisDecorations)) {
+    if (inst->IsDecoration()) {
+      get_decoration_mgr()->AddDecoration(inst);
+    }
+  }
+}
+
+void IRContext::KillNamesAndDecorates(uint32_t id) {
+  std::vector<ir::Instruction*> decorations =
+      get_decoration_mgr()->GetDecorationsFor(id, true);
+
+  for (Instruction* inst : decorations) {
+    KillInst(inst);
+  }
+
+  // Should the decoration manger take care of the SpvOpName Instruction as
```

Comment:
An OpName is not a decoration. Decorations are semantically significant but OpName is a debug instruction without semantic signficance.

I think this is the right place to delete the name.   It's right in the method name too.

"manger" -> "manager".

Once we agree, either delete this comment or put in the rationale for doing the name deletion here.

Reply:
OK, but shouldn't this also handle SpvOpMemberName then?

Reply:
Probably.  OpMemberName is a little different because it gets applied to members of a type, the specific applications where this function was used before did not care about removing types, so it was not an issue.

I can add it in.

Reply:
I think I agree with @dneto0  that the name does not have semantic significance.  This will create special cases in the decoration manager when we ask if two ids have the same decorations.  Does that include the name?  I'll keep the removal of the names this way.